src/main/java/org/prebid/server/settings/HttpApplicationSettings.java

```diff
@@ -77,12 +78,61 @@ public HttpApplicationSettings(HttpClient httpClient, JacksonMapper mapper, Stri
         this.videoEndpoint = HttpUtil.validateUrl(Objects.requireNonNull(videoEndpoint));
     }
 
-    /**
-     * Not supported and returns failed result.
-     */
     @Override
     public Future<Account> getAccountById(String accountId, Timeout timeout) {
-        return Future.failedFuture(new PreBidException("Not supported"));
+
+        return fetchAccountsByIds(Collections.singleton(accountId), timeout)
+                .map(accounts -> accounts.stream()
+                        .findFirst()
+                        .orElseThrow(() ->
+                                new PreBidException(String.format("Account with id :%s not found", accountId))));
+    }
+
+    private Future<Set<Account>> fetchAccountsByIds(Set<String> accountIds, Timeout timeout) {
+        if (CollectionUtils.isEmpty(accountIds)) {
+            return Future.succeededFuture(Collections.emptySet());
+        }
+        final long remainingTimeout = timeout.remaining();
+        if (timeout.remaining() <= 0) {
+            return Future.failedFuture(new TimeoutException("Timeout has been exceeded"));
+        }
+
+        return httpClient.get(accountsRequestUrlFrom(endpoint, accountIds), HttpUtil.headers(), remainingTimeout)
+                .compose(response -> processAccountsResponse(response, accountIds))
+                .recover(Future::failedFuture);
+    }
+
+    private static String accountsRequestUrlFrom(String endpoint, Set<String> accountIds) {
+        final StringBuilder url = new StringBuilder(endpoint);
+        url.append(endpoint.contains("?") ? "&" : "?");
+
+        if (!accountIds.isEmpty()) {
+            url.append("account-ids=[\"").append(joinIds(accountIds)).append("\"]");
+        }
+
+        return url.toString();
+    }
+
+    private Future<Set<Account>> processAccountsResponse(HttpClientResponse response, Set<String> accountIds) {
+        return Future.succeededFuture(
+                toAccountsResult(response.getStatusCode(), response.getBody(), accountIds));
+    }
+
+    private Set<Account> toAccountsResult(int statusCode, String body, Set<String> accountIds) {
+        if (statusCode != HttpResponseStatus.OK.code()) {
+            throw new PreBidException(String.format("Error fetching accounts %s via http: "
+                    + "unexpected response status %d", accountIds, statusCode));
+        }
+
+        final HttpAccountsResponse response;
+        try {
+            response = mapper.decodeValue(body, HttpAccountsResponse.class);
+        } catch (DecodeException e) {
+            throw new PreBidException(String.format("Error fetching accounts %s "
+                    + "via http: failed to parse response: %s", accountIds, e.getMessage()));
+        }
+
+        return new HashSet<>(response.getAccounts().values());
```

Comment:
Potentially, accounts can be undefined in response, so we should tolerate the case.